source/base/NexusApp.cc

```diff
@@ -76,27 +79,27 @@ NexusApp::NexusApp(G4String init_macro): G4RunManager(), gen_name_(""),
   // The physics lists are handled with Geant4's own 'factory'
   physicsList = new G4GenericPhysicsList();
 
-  batch_ = std::make_unique<BatchSession>(init_macro.c_str());
-  batch_->SessionStart();
+  auto batch = std::make_unique<BatchSession>(init_macro.c_str());
+  batch->SessionStart();
```

Comment:
+ `batch` is a `unique_ptr<BatchSession>` therefore it owns the `BatchSession`.
+ It is also a *local* variable, therefore it will go out of scope as soon as we reach the end of this function.
+ Ownership is not transferred anywhere before we reach the end of this function.
+ Ergo, the `BatchSession` gets deleted as soon as we reach the end of this function (`NexusApp::NexusApp`).

Now, maybe that's what is supposed to happen (I don't have the time try to follow the logic of `BatchSession`) but it does seem a bit odd and forces the reader to hunt for uses of `batch` in this veeery long function.

Which is why I *thoroughly* recommend that you use local scopes to limit lifetimes: put these two lines between a pair of `{ ... }` and it becomes immediately obvious that `batch` will not be used again after these two lines. It *really* does simplify reading the code.

Once you've done that, you might wonder, "Why don't I just write it as `BatchSession(init_macro.c_str()).SessionStart()`?

Valid answers might be

+ The average Nexus contributor might not spot the full implications of this line: A temporary, stack-allocated `BatchSession` is automatically destructed as soon as `SessionStart()` returns.

+ The smart pointer version suggests to the reader of the code that the author of the code has actually thought about lifetimes here, rather than being completely oblivious to the need to think about them.

As much as I believe that you should write as few comments as possible, because you should write code that is self-explanatory, in this case I would recommend a comment. You are programming in a context where you cannot assume that the people who will read and modify this code have a sufficiently good understanding of the lifetime implications. Consequently, code which is self-explanatory to a C++ expert, will remain mysterious to  many Nexus contributors. Write a comment which points out that/where the `BatchSession` is destructed.

And *do* limit the scope of variables such as these, using anonymous scopes. That way the aforementioned comment will be here, rather than 70 lines further down.